src/pages/workspaces/workspace/Analyses.js

```diff
@@ -258,11 +257,6 @@ const Analyses = _.flow(
     withErrorReporting('Error loading analyses'),
     Utils.withBusyState(setBusy)
   )(async () => {
```

Comment:
This may cause a change in behavior that might not be desirable with regard to "requester pays" workspaces. The call in `WorkspaceContainer` to obtain the location/locationType does **not** use `withRequestPaysHandler` (nor do we want it to, as that would force the user to select a billing workspace whenever any part of the Workspace was rendered); in the case of a requester pays workspace, a 400 error will result and location/locationType will not be populated.

When you remove these extra calls to `checkBucketLocation`, please make sure that there is still handling in each location that will trigger the "requester pays" logic appropriately, and that the values then populate after the user selects a workspace to bill to.

Reply:
It's even worse than that in WorkspaceContainer... `loadBucketLocation` has a conditional on `isGoogleWorkspace`.

https://github.com/DataBiosphere/terra-ui/blob/5e451b1699d22e1f7f287c7fd724fff74790ac2c/src/pages/workspaces/workspace/WorkspaceContainer.js#L312-L317

`isGoogleWorkspace` is based on the `googleProject` state variable, which is initialized from the `workspace` state variable. `workspace` is undefined until the workspace is loaded in `refreshWorkspace`. 

https://github.com/DataBiosphere/terra-ui/blob/5e451b1699d22e1f7f287c7fd724fff74790ac2c/src/pages/workspaces/workspace/WorkspaceContainer.js#L291

So on the first render, `isGoogleWorkspace` is false and `loadBucketLocation` captures that value. Thus, `wrapWorkspace` won't actually load the bucket location on mount. It will only load after something calls `refreshWorkspace` again.

Reply:
The call to get a bucket location will fail for Azure workspaces, so we do need to delay the call until we know the type of workspace. It seems like we could stop calling loadBucketLocation in onMount.

Reply:
> The call to get a bucket location will fail for Azure workspaces, so we do need to delay the call until we know the type of workspace.

That makes sense. And `loadBucketLocation` is called after the workspace is fetched. The issue with the current implementation of `loadBucketLocation` closes over an `isGoogleWorkspace` variable that based on the `workspace` state variable instead of the workspace that was fetched in `refreshWorkspace`.

Reply:
@jdcanas this location does have handling for request pays-- see https://github.com/DataBiosphere/terra-ui/pull/3037/files#diff-e19c52f5a9d3432ff7537aebcbb3d2a439a5fefc7346c1b0ac0017e2f9e8652bR256

In contrast, the call in WorkspaceContainer to obtain the location/locationType does not use withRequestPaysHandler (nor do we want it to, as that would force the user to select a billing workspace whenever any part of the Workspace was rendered); in the case of a requester pays workspace, a 400 error will result and location/locationType will not be populated.

@nawatts Can you share a requester pays workspace on dev with both me and Justin?

Reply:
@jdcanas while you are waiting for a requester pays workspace on dev, here is one in prod so you can get an idea of what the experience is https://app.terra.bio/#workspaces/anvil-datastorage/AnVIL_T2T 

Note though that the requirement to choose a billing workspace on the workspace dashboard is going away with https://github.com/DataBiosphere/terra-ui/pull/3045

Reply:
> the main goal here is that I believe location should be resolved at the workspace level, as opposed to the API call being scattered across all components that rely on it.

I would agree, with some caveats. In the long term, this will probably be solved by having bucket location be part of the workspace data model in Rawls where we can count on it always being available with the rest of the workspace information. In the short term, different parts of the workspace UI have different needs for the bucket location. The workspace dashboard displays the bucket location but should still be viewable even if the request for it fails because the bucket is requester pays. I'm guessing the analyses tab has a harder requirement for the bucket location? Those differing requirements complicate our ability to handle requester pays buckets at the workspace level in such a way that child components could rely on having the location available.

There is another nuance of the requester pays handling that may help with loading bucket location within analyses components. The user only has to choose a workspace to bill to in cases where they have read-only access to the current workspace. If they can compute on the current workspace or own it, requester pays requests are automatically billed to the current workspace and the request for bucket location succeeds. If analyses components are only reachable when a user can compute on the workspace, they can reliably load the bucket location.

https://github.com/DataBiosphere/terra-ui/blob/7aea4d71421d4ab4d31c999fe4ba0ae1a27b7cb8/src/libs/ajax.js#L99-L104

https://github.com/DataBiosphere/terra-ui/blob/7aea4d71421d4ab4d31c999fe4ba0ae1a27b7cb8/src/libs/ajax.js#L115-L117